src/App.tsx

```diff
@@ -17,18 +38,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filteredBy={filteredBy}
+                onFilteredBy={setFilteredBy}
+                query={query}
+                onQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && !selectedTodo && (
+                <Loader />
+              )}
+
+              {isTodoList && (
+                <TodoList
+                  todos={vidibleTodos}
+                  onSelect={(todo) => setSelectedTodo(todo)}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
```

Comment:
Please prevent modal window flicker (it feels like when you click on the button, it appears for a moment)
https://www.loom.com/share/3619273481a24b21a37c6d4e9aa4eb57?sid=cddf6b85-4210-4494-81dd-753515d87565